bootstrap

```diff
@@ -197,7 +193,6 @@ ${cmake} -DCMAKE_BUILD_TYPE=${build_type} \
     -DTILEDB_S3=${tiledb_s3} \
     -DTILEDB_AZURE=${tiledb_azure} \
     -DTILEDB_GCS=${tiledb_gcs} \
-    -DTILEDB_MEMFS=${tiledb_memfs} \
```

Comment:
Since we are removing this config variable to have Memfs always on, let's clean it up from those files as well:
```
cmake/TileDB-Superbuild.cmake
test/CMakeLists.txt
tiledb/CMakeLists.txt
scripts/azure-linux_mac.yml
```